Purchases/Attribution/AttributionFetcher.swift

```diff
@@ -138,4 +134,56 @@ private extension AttributionFetcher {
         }
     }
 
+    @available(iOS 14.0.0, tvOS 14.0.0, *)
+    private var fetchAuthorizationStatus: FakeTrackingManagerAuthorizationStatus {
+        let needsTrackingAuthorization = self.needsTrackingAuthorization
+
+        guard let trackingManagerProxy = self.trackingProxy else {
+            if needsTrackingAuthorization {
+                return .denied
+            } else {
+                return .notDetermined
+            }
+        }
+
+        let authStatusSelector = NSSelectorFromString(trackingManagerProxy.authorizationStatusPropertyName)
+        guard trackingManagerProxy.responds(to: authStatusSelector) else {
+            Logger.warn(Strings.attribution.att_framework_present_but_couldnt_call_tracking_authorization_status)
+            return .denied
+        }
+
+        let authStatus = callAuthStatusSelector(authStatusSelector, trackingManagerProxy: trackingManagerProxy)
+        return authStatus
+    }
+
+    private var trackingProxy: TrackingManagerProxy? {
+        let trackingManagerProxy = attributionFactory.atFollowingProxy()
+        if trackingManagerProxy == nil && needsTrackingAuthorization {
+            Logger.warn(Strings.attribution.search_ads_attribution_cancelled_missing_att_framework)
+        }
+        return trackingManagerProxy
+    }
+
+    private var needsTrackingAuthorization: Bool {
+        let minimumOSVersionRequiringAuthorization = OperatingSystemVersion(majorVersion: 14,
+                                                                            minorVersion: 5,
+                                                                            patchVersion: 0)
+        let needsTrackingAuthorization = systemInfo
+            .isOperatingSystemAtLeastVersion(minimumOSVersionRequiringAuthorization)
+        return needsTrackingAuthorization
```

Comment:
Nit: this has the same name as the property itself, so I think returning
```swift
systemInfo
             .isOperatingSystemAtLeastVersion(minimumOSVersionRequiringAuthorization)
```
directly would be just as clear.

Reply:
✅